src/compiler/checker.ts

```diff
@@ -3174,7 +3174,7 @@ namespace ts {
              * ensuring that any names written with literals use element accesses.
              */
             function appendPropertyOrElementAccessForSymbol(symbol: Symbol, writer: SymbolWriter): void {
-                const symbolName = getNameOfSymbol(symbol);
+                const symbolName = symbol.escapedName === "default" ? "default" : getNameOfSymbol(symbol);

```

Comment:
This feels somewhat unfortunate/ad-hoc. Under which use-cases did we want the internal name of the exported symbol versus the `default` symbol name?

If this is how this is going to be fixed, can `getNameOfSymbol` be documented in a comment saying that it returns the internal symbol name for a `default` exported symbol, along with any other interesting special cases it performs? (This way when it is used hopefully we remember to check if all those cases are OK in that scenario)